api/src/main/java/org/openmrs/ui/framework/FormatterImpl.java

```diff
@@ -42,6 +39,7 @@
 
     private MessageSource messageSource;
     private AdministrationService administrationService;
+    private UiUtils ui = new BasicUiUtils();

```

Comment:
@mogoodrich is it ok to access `UiUtils` from here and in that way?
I would think that this was not the intent of this class, but maybe it's ok and since it works...

Reply:
In general, I'm okay with it, but isn't a bit weird/recursive in this case, because the UiUtils has the formatter wired into it?   ie the format method below is usually accessed via "uiUtils.format()" ?

Reply:
We can always remove that UiUtils and get the GP directly... Instead of using ui.handleTimeZones().